### DISCUSSION TITLE
how to enable local insecure docker registry on talos

### DISCUSSION THREAD

**Discussion by yingwana3** [siderolabs/talos#7535] [Q&A] [answered]
Hi

We want to pull images (which are not open source) from our local insecure docker registry (10.67.x.x/5000) on talos but don't know how.

It seems controlplane.yaml has something with registry configuration as following 

 registries: {}
    # # Specifies mirror configuration for each registry host namespace.
    # mirrors:
    #     ghcr.io:
    #         # List of endpoints (URLs) for registry mirrors to use.
    #         endpoints:
    #             - https://registry.insecure
    #             - https://ghcr.io/v2/

    # # Specifies TLS & auth configuration for HTTPS image registries.
    # config:
    #     registry.insecure:
    #         # The TLS configuration for the registry.
    #         tls:
    #             insecureSkipVerify: true # Skip TLS server certificate verification (not recommended).
    #
    #             # # Enable mutual TLS authentication with the registry.
    #             # clientIdentity:
    #             #     crt: LS0tIEVYQU1QTEUgQ0VSVElGSUNBVEUgLS0t
    #             #     key: LS0tIEVYQU1QTEUgS0VZIC0tLQ==
    #
    #         # # The auth configuration for this registry.
    #         # auth:
    #         #     username: username # Optional registry authentication.
    #         #     password: password # Optional registry authentication.
But we don't know how to modify them to adapt to our local docker registry

**Comment by smira** [comment, marked as answer]
Please see examples in the docs:

* https://www.talos.dev/v1.4/talos-guides/configuration/pull-through-cache/
* https://www.talos.dev/v1.4/advanced/air-gapped/#closing-notes
* reference: https://www.talos.dev/v1.4/reference/configuration/#registriesconfig

**Reply by yingwana3** [reply]
Hi
I have followed this guide https://www.talos.dev/v1.4/advanced/air-gapped/#closing-notes to first setup internal registry.
curl http://127.0.0.1:6000/v2/_catalog
{"repositories":["coredns/coredns","etcd-development/etcd","flexran_vdu","kube-apiserver","kube-controller-manager","kube-proxy","kube-scheduler","pause","siderolabs/flannel","siderolabs/install-cni","siderolabs/installer","siderolabs/kubelet"]}
flexran_vdu is the needed image i push to the registry.

Our talos install config is like below:
![image](https://github.com/siderolabs/talos/assets/60091241/0695956c-edc2-40bd-b54e-3833fc907f48)

I modify registry in controlplane.yaml like below：
![image](https://github.com/siderolabs/talos/assets/60091241/0d7b3737-4cc7-4b13-8e44-8a223a75fb66)


Then I create a pod on talos to pull the flexran_vdu image but failed with folloing logs:

![image](https://github.com/siderolabs/talos/assets/60091241/c1dd0c44-1e1f-4741-bf88-66f138e1d574)

**Reply by yingwana3** [reply]
we install talos on baremetal platform with *.iso.

**Reply by yingwana3** [reply]
we have successfully configure the https registry, so please ignore the above question.

**Reply by smira** [reply]
Talos did the job above - the container request was redirected, but your registry is down.
